src/predict_api/flask_app.py

```diff
@@ -0,0 +1,243 @@
+import json
+
+import requests
+from apispec import APISpec
+from apispec_webframeworks.flask import FlaskPlugin
+from apispec.ext.marshmallow import MarshmallowPlugin
+from dotenv import load_dotenv
+from flask import Flask, g, redirect, request, url_for, jsonify
+from flask_login import (
+    LoginManager,
+    login_required,
+    login_user,
+    logout_user,
+)
+from flask_cors import CORS
+from flask_sqlalchemy import SQLAlchemy
+from oauthlib.oauth2 import WebApplicationClient
+
+from config import Config, logging
+
+from flask import jsonify, render_template, Blueprint
+from flask_login import login_required
+
+from config import logging, Config
+
+
+
+# TODO: many below
+# figure out topology. This should run in a separate container. Are requests all proxied through the annotation api?
+# auth - either it runs behind the main container or we need separate auth
+# CORS lockdown - we're currently allowing any cross-origin request
+# options to add to the API:
+#   - return diagnostic on processing time
+#   - versions of the inference algo - over time we want this more robust
+# model versioning.  Maybe the model comes from a blob storage url?
```

Comment:
I'm not sure of the AzureML "way", but MLFlow registry offers a standard that might be useful here https://mlflow.org/docs/latest/model-registry.html?

That would combine blob storage with model metadata in the DB (we could use the same one as the annotation API to keep infrastruture costs down) https://mlflow.org/docs/latest/tracking.html#backend-stores